src/cbuild/hooks/do_pkg/999_gen_apk.py

```diff
@@ -50,35 +50,38 @@ def genpkg(pkg, repo, arch, binpkg):
     if pkg.rparent.git_revision and not pkg.rparent.git_dirty:
         pargs += ["--info", f"repo-commit:{pkg.rparent.git_revision}"]
 
-    # dependencies of any sort
-    deps = []
-
-    # bootstrap packages are not installable ootb
-    if pkg.pkgname.endswith("-bootstrap") and pkg.build_style != "meta":
-        deps += ["bootstrap:" + pkg.pkgname.removesuffix("-bootstrap")]
-
-    # explicit package depends
-    for c in pkg.depends:
-        ploc = c.find("!")
-        if ploc > 0:
-            deps.append(c[0:ploc].removeprefix("virtual:"))
-        else:
-            deps.append(c.removeprefix("virtual:"))
-
-    # sort before adding more
-    deps.sort()
-
-    # shlib requires
-    if hasattr(pkg, "so_requires"):
-        pkg.so_requires.sort()
-        deps += map(lambda v: f"so:{v}", sorted(pkg.so_requires))
-
-    # .pc file requires
-    if hasattr(pkg, "pc_requires"):
-        deps += map(lambda v: f"pc:{v}", sorted(pkg.pc_requires))
-
-    if len(deps) > 0:
-        pargs += ["--info", f"depends:{' '.join(deps)}"]
+    if not hasattr(pkg, "all_rdeps"):
+        # dependencies of any sort
+        deps = []
+
+        # bootstrap packages are not installable ootb
+        if pkg.pkgname.endswith("-bootstrap") and pkg.build_style != "meta":
+            deps += ["bootstrap:" + pkg.pkgname.removesuffix("-bootstrap")]
+
+        # explicit package depends
+        for c in pkg.depends:
+            ploc = c.find("!")
+            if ploc > 0:
+                deps.append(c[0:ploc].removeprefix("virtual:"))
+            else:
+                deps.append(c.removeprefix("virtual:"))
+
+        # sort before adding more
+        deps.sort()
+
+        # shlib requires
+        if hasattr(pkg, "so_requires"):
+            pkg.so_requires.sort()
+            deps += map(lambda v: f"so:{v}", sorted(pkg.so_requires))
+
+        # .pc file requires
+        if hasattr(pkg, "pc_requires"):
+            deps += map(lambda v: f"pc:{v}", sorted(pkg.pc_requires))
+
+        pkg.all_rdeps = deps
```

Comment:
this is the same as `collect_deps` in `000_collect_rdeps.py`, what would be an appropriate place to put that function so it can be reused here? `cbuild.util`?